### PR TITLE
hostsubnet requires a specific annotation for multi-tenancy to work

### PR DESCRIPTION
**Pull request by rajatchopra**
Missed the annotation in the code blob. Without this fix, multi-tenancy will block F5 from operating across projects.

@lihongan Review please. Thanks.
/cc @ahardin-rh

### CODE REVIEW

**Comment by lihongan** [comment]
Yes, the annotation `pod.network.openshift.io/fixed-vnid-host: "0"` is required when creating the hostsubnet for F5 server if using multitenant plugin.

**Comment by lihongan** [comment]
and the related PR is: https://github.com/openshift/origin/pull/11817

**Comment by lihongan** [comment]
@rajatchopra Didnot see below privilege description in the doc but it is also required.
`oadm policy add-cluster-role-to-user system:sdn-reader system:serviceaccount:default:router`

Please confirm, thanks

File: architecture/core_concepts/routes.adoc [hunk 1]
```diff
@@ -427,7 +427,9 @@ address. All nodes of the cluster must learn this automatically.
 . Since the overlay network is multi-tenant aware, F5 must use a VxLAN ID that is
 representative of an `admin` domain, ensuring that all tenants are reachable by
 the F5. Ensure that F5 encapsulates all packets with a `vnid` of `0` (the
-default `vnid` for the `admin` name space in {product-title}).
+default `vnid` for the `admin` name space in {product-title}). This is accomplished
+by putting an annotation on the manually created hostsubnet -
```
Comment (ahardin-rh) [inline architecture/core_concepts/routes.adoc]: `hostsubnet`

File: architecture/core_concepts/routes.adoc [hunk 2]
```diff
@@ -427,7 +427,9 @@ address. All nodes of the cluster must learn this automatically.
 . Since the overlay network is multi-tenant aware, F5 must use a VxLAN ID that is
 representative of an `admin` domain, ensuring that all tenants are reachable by
 the F5. Ensure that F5 encapsulates all packets with a `vnid` of `0` (the
-default `vnid` for the `admin` name space in {product-title}).
+default `vnid` for the `admin` name space in {product-title}). This is accomplished
+by putting an annotation on the manually created hostsubnet - 
+pod.network.openshift.io/fixed-vnid-host: 0
```
Comment (ahardin-rh) [inline architecture/core_concepts/routes.adoc]: `pod.network.openshift.io/fixed-vnid-host: 0`

**Comment by rajatchopra** [comment]
@lihongan Thanks. Yes we need that capability also.
@ahardin-rh Fixed according to feedback. Please merge if you think its ready.
Thanks.

**Comment by ahardin-rh** [comment]
@rajatchopra Thanks! Looks like this is targeted for 3.4. Is that correct?

**Comment by rajatchopra** [comment]
@ahardin-rh Correct. Needs a patch to 3.4, is there a branch I should port this PR to?

**Comment by ahardin-rh** [comment]
@rajatchopra Thanks! I will merge this (into master), then our team will cherry-pick it to the appropriate branches (per the labels) during our next weekly release. So, no further action required from you. Thanks again!

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../architecture/core_concepts/routes.adoc#architecture-core-concepts-routes[Core Concepts -> Routes]
|Clarified that `hostsubnet` requires a specific annotation for multi-tenancy to work.
%
|xref:../install_config/router/f5_router.adoc#install-config-router-f5[Setting up a Router -> Using the F5 Router Plug-in]
|Clarified that `hostsubnet` requires a specific annotation for multi-tenancy to work.
%

**Pull request merged**